pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -622,24 +640,33 @@ private void searchForCandidateBroker(NamespaceBundle bundle,
     }
 
     private boolean isBrokerActive(String candidateBroker) {
-        List<String> brokers = pulsar.getLocalMetadataStore().getChildren(LoadManager.LOADBALANCE_BROKERS_ROOT).join();
-
-        for (String brokerHostPort : brokers) {
-            if (candidateBroker.equals("http://" + brokerHostPort)) {
+        URI uri = URI.create(candidateBroker);
+        String candidateBrokerHostAndPort = uri.getHost() + ":" + uri.getPort();
+        Set<String> availableBrokers = getAvailableBrokers();
+        for (String brokerHostPort : availableBrokers) {
+            if (candidateBrokerHostAndPort.equals(brokerHostPort)) {
                 if (LOG.isDebugEnabled()) {
-                    LOG.debug("Broker {} found for SLA Monitoring Namespace", brokerHostPort);
+                    LOG.debug("Broker {} is available.", brokerHostPort);
                 }
                 return true;
             }
         }
```

Comment:
Nit: we could use `availableBrokers.contains(candidateBrokerHostAndPort)` here, which could be a minor optimization for clusters with many brokers and many topics getting loaded.

Reply:
Makes sense. I missed that opportunity in refactoring

Reply:
done